### PR TITLE
DOC: add example for sparse.density and sparse.coo

### PR DESCRIPTION
**Pull request by Moisan**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/sparse/accessor.py [hunk 1]
```diff
@@ -329,6 +329,18 @@ def to_coo(self):
...
+        --------
+        >>> df = pd.DataFrame({"A": pd.arrays.SparseArray([0, 1, 0, 1])})
+        >>> df.sparse.to_coo()
+        <4x1 sparse matrix of type '<class 'numpy.int64'>'
+                with 2 stored elements in COOrdinate format>
+        >>> sparse_array = pd.arrays.SparseArray([0, 1.234, 0, 8.32], fill_value=0)
+        >>> df = pd.DataFrame({"A": sparse_array})
+        >>> df.sparse.to_coo()
+        <4x1 sparse matrix of type '<class 'numpy.float64'>'
+                with 2 stored elements in COOrdinate format>
```
Comment (MarcoGorelli) [inline pandas/core/arrays/sparse/accessor.py:343]: thanks for working on this - does `fill_value` do anything here?

I'd suggest either:
- writing an example where `fill_value` has an effect
- just keeping the example above

Comment (Moisan) [reply pandas/core/arrays/sparse/accessor.py:343]: `to_coo()` requires the `fill_value` to be zero, but when using floats, it is [`np.nan` by default](https://pandas.pydata.org/docs/reference/api/pandas.arrays.SparseArray.html):

```
>>> df = pd.DataFrame({"A": pd.arrays.SparseArray([0, 1.234, 0, 8.32])})
>>> df.sparse.to_coo()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/thierrymoisan/dev/pandas/pandas/core/arrays/sparse/accessor.py", line 344, in to_coo
    raise ValueError("fill value must be 0 when converting to COO matrix")
ValueError: fill value must be 0 when converting to COO matrix
```

So I  believe `fill_value` does have an effect in this example.

Comment (MarcoGorelli) [reply pandas/core/arrays/sparse/accessor.py:343]: thanks for explaining - to be honest I'd leave this second example out, I'm not sure what it adds (either than, or add a sentence before the example explaining what it does)

**Review by MarcoGorelli — APPROVED** [review]
Nice, thanks @Moisan !

**Pull request merged**